cmd/frontend/graphqlbackend/campaigns.go

```diff
@@ -92,97 +51,64 @@ type FileDiffsConnectionArgs struct {
-	return nil, campaignsOnlyInEnterprise
-}
-
-func (defaultCampaignsResolver) DeleteCampaign(ctx context.Context, args *DeleteCampaignArgs) (*EmptyResponse, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+type CampaignsResolver interface {
+	// Mutations
+	CreateCampaign(ctx context.Context, args *CreateCampaignArgs) (CampaignResolver, error)
+	ApplyCampaign(ctx context.Context, args *ApplyCampaignArgs) (CampaignResolver, error)
+	MoveCampaign(ctx context.Context, args *MoveCampaignArgs) (CampaignResolver, error)
+	CloseCampaign(ctx context.Context, args *CloseCampaignArgs) (CampaignResolver, error)
+	DeleteCampaign(ctx context.Context, args *DeleteCampaignArgs) (*EmptyResponse, error)
+	CreateChangesetSpec(ctx context.Context, args *CreateChangesetSpecArgs) (ChangesetSpecResolver, error)
+	CreateCampaignSpec(ctx context.Context, args *CreateCampaignSpecArgs) (CampaignSpecResolver, error)
+	// ComputeCampaignDelta(ctx context.Context, args *ComputeCampaignDeltaArgs) (CampaignDeltaResolver, error)
+	SyncChangeset(ctx context.Context, args *SyncChangesetArgs) (*EmptyResponse, error)
 
-func (defaultCampaignsResolver) RetryCampaignChangesets(ctx context.Context, args *RetryCampaignChangesetsArgs) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
+	// Queries
+	Campaigns(ctx context.Context, args *ListCampaignArgs) (CampaignsConnectionResolver, error)
+	CampaignByID(ctx context.Context, id graphql.ID) (CampaignResolver, error)
+	ChangesetByID(ctx context.Context, id graphql.ID) (ChangesetResolver, error)
 }
 
-func (defaultCampaignsResolver) CloseCampaign(ctx context.Context, args *CloseCampaignArgs) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+type CampaignSpecResolver interface {
+	ID() (graphql.ID, error)
 
-func (defaultCampaignsResolver) PublishCampaignChangesets(ctx context.Context, args *PublishCampaignChangesetsArgs) (CampaignResolver, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+	OriginalInput() (string, error)
+	ParsedInput() (JSONValue, error)
+	ChangesetSpecs() ([]ChangesetSpecResolver, error)
 
-func (defaultCampaignsResolver) PublishChangeset(ctx context.Context, args *PublishChangesetArgs) (*EmptyResponse, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+	Creator() (*UserResolver, error)
+	CreatedAt() *DateTime
+	Namespace() (*NamespaceResolver, error)
 
-func (defaultCampaignsResolver) SyncChangeset(ctx context.Context, args *SyncChangesetArgs) (*EmptyResponse, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+	ExpiresAt() *DateTime
 
-func (defaultCampaignsResolver) CreateChangesets(ctx context.Context, args *CreateChangesetsArgs) ([]ExternalChangesetResolver, error) {
-	return nil, campaignsOnlyInEnterprise
+	PreviewURL() (string, error)
 }
 
-func (defaultCampaignsResolver) ChangesetByID(ctx context.Context, id graphql.ID) (ChangesetResolver, error) {
-	return nil, campaignsOnlyInEnterprise
-}
+type ChangesetSpecResolver interface {
+	ID() (graphql.ID, error)
```

Comment:
I think this should be, otherwise the Node interface is not fulfilled

```suggestion
	ID() graphql.ID
```